docs/architecture/grpc-for-wcf-developers/Protocol Buffers.md

```diff
@@ -0,0 +1,339 @@
+| ------------- | ---------- | ----- |
+| double        | double     |       |
+| float         | float      |       |
+| int32         | int        |       |
+| int64         | long       |       |
+| uint32        | uint       |       |
+| uint64        | ulong      |       |
+| sint32        | int        | 1     |
+| sint64        | long       | 1     |
+| fixed32       | uint       | 2     |
+| fixed64       | ulong      | 2     |
+| sfixed32      | int        | 2     |
+| sfixed64      | long       | 2     |
+| bool          | bool       |       |
+| string        | string     | 3     |
+| bytes         | ByteString | 4     |
+
+
+### Notes
+
+1. The standard encoding for `int32` and `int64` is inefficient when working with signed values. If your field is likely to contain negative numbers, use `sint32` or `sint64` instead. Both types map to the C# `int` and `long` types respectively.
+2. The `fixed` fields always use the same number of bytes regardless of the value. This makes serialization/deserialization faster for larger values.
+3. Protobuf strings are UTF-8 (or 7-bit ASCII) encoded, and the encoded length cannot be greater than 2<sup>32</sup>.
+4. The Protobuf runtime provides a `ByteString` type that maps easily to and from C# `byte[]` arrays.
+
+## Dates and Times
+
+The native scalar types do not provide for date and time values, equivalent to C#'s `DateTimeOffset`, `DateTime` and `TimeSpan`. These types can be specified using some of Google's "Well Known Types" extensions, which provide code generation and runtime support for more complex field types across the supported platforms. To use them, you need to import them in your `.proto` file, like this:
+
+```protobuf  
+syntax = "proto3"
+
+import "google/protobuf/duration.proto";  
+import "google/protobuf/timestamp.proto";
+
+message Meeting {
+
+    string subject = 1;
+    google.protobuf.Timestamp time = 2;
+    google.protobuf.Duration duration = 3;
+
+}  
+```
+
+## Nullable Types
+
+The Protobuf code generation for C# will use the native types, such as `int` for `int32`. This means that the values are always included and can't be null. For values that require explicit null, where you would use e.g. `int?` in your C# code, Protobuf's "Well Known Types" include wrappers that are compiled to nullable C# types. To use them, import `wrappers.proto` into your `.proto` file, like this:
+
+```protobuf  
+syntax = "proto3"
+
+import "google/protobuf/wrappers.proto"
+
+message Person {
+
+    ...
+    google.protobuf.Int32Value age = 5;
+
+}
+```
+
+Here is the complete list of wrapper types with their equivalent C# type:
+
+| C# Type | Well Known Type wrapper     |
+| ------- | --------------------------- |
+| double? | google.protobuf.DoubleValue |
+| float?  | google.protobuf.FloatValue  |
+| int?    | google.protobuf.Int32Value  |
+| long?   | google.protobuf.Int64Value  |
+| uint?   | google.protobuf.UInt32Value |
+| ulong?  | google.protobuf.UInt64Value |
+
+## ??? Decimals ???
```

Comment:
Is this PR still a WIP? Or do you want to omit this for now?

Reply:
Leave it for now. There's something up with git, I'm going to have to completely refork the repo.